File: 프로그래머스/lv2/12985. 예상 대진표/예상 대진표.py
def solution(n,a,b):
    answer = 0

    while n > 1:
        # print(n, a, b)
        answer += 1
        for i in range(0, n, 2):    
            #print(i)
            p1, p2 = i+1, i+2
            if (p1 == a and p2 == b) or (p1 == b and p2 == a):
                return answer
            elif p1 == a or p2 == a:
                a = i//2 +1
            elif p1 == b or p2 == b:
                b = i//2 +1
        n = n // 2
                
                
        

    return answer

File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.java
class Solution
{
    public int solution(int n, int a, int b)
    {
        if(n == 2) return 1;
        int answer = 0;                
        
        // a와 b가 같아지면 경기에서 만난다는 의미
        while (a != b) {
            // (번호 + 1) / 2를 하면 다음 라운드 번호가 됩니다.            
            a = (a + 1) / 2;
            b = (b + 1) / 2;
            answer++;
        }
        
        return answer;
    }
}